File: solutions/_2215_find_the_difference_of_two_arrays/solution.java
package _2215_find_the_difference_of_two_arrays;

import java.util.*;

class Solution {

  public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
    Set<Integer> set1 = new HashSet<>();
    Set<Integer> set2 = new HashSet<>();

    for (int n : nums1) set1.add(n);
    for (int n : nums2) set2.add(n);

    List<List<Integer>> res = new ArrayList<>();
    res.add(new ArrayList<>());
    res.add(new ArrayList<>());

    set1
      .iterator()
      .forEachRemaining(n -> {
        if (!set2.contains(n)) {
          res.get(0).add(n);
        }
      });

    set2
      .iterator()
      .forEachRemaining(n -> {
        if (!set1.contains(n)) {
          res.get(1).add(n);
        }
      });

    return res;
  }
}
/*
https://leetcode.com/problems/find-the-difference-of-two-arrays/
*/

File: solutions/_2215_find_the_difference_of_two_arrays/solution.py
from typing import List


class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1 = set(nums1)
        s2 = set(nums2)

        return [[n for n in s1 if n not in s2], [n for n in s2 if n not in s1]]


# https://leetcode.com/problems/find-the-difference-of-two-arrays/
